dpctl/_sycl_queue.pyx

```diff
@@ -121,6 +122,47 @@ cdef class kernel_arg_type_attribute:
         return self.attr_value
 
 
+cdef class LocalAccessor:
+    cdef _md_local_accessor lacc
+
+    def __cinit__(self, size_t ndim, str type, size_t dim0, size_t dim1, size_t dim2):
+       self.lacc.ndim = ndim
+       self.lacc.dim0 = dim0
+       self.lacc.dim1 = dim1
+       self.lacc.dim2 = dim2
+
+       if ndim < 1 or ndim > 3:
+           raise ValueError
+       if type == 'i1':
+           self.lacc.dpctl_type_id = _arg_data_type._INT8_T
+       elif type == 'u1':
+           self.lacc.dpctl_type_id = _arg_data_type._UINT8_T
+       elif type == 'i2':
+           self.lacc.dpctl_type_id = _arg_data_type._INT16_T
+       elif type == 'u2':
+           self.lacc.dpctl_type_id = _arg_data_type._UINT16_T
+       elif type == 'i4':
+           self.lacc.dpctl_type_id = _arg_data_type._INT32_T
+       elif type == 'u4':
+           self.lacc.dpctl_type_id = _arg_data_type._UINT32_T
+       elif type == 'i8':
+           self.lacc.dpctl_type_id = _arg_data_type._INT64_T
+       elif type == 'u8':
+           self.lacc.dpctl_type_id = _arg_data_type._UINT64_T
+       elif type == 'f4':
+           self.lacc.dpctl_type_id = _arg_data_type._FLOAT
+       elif type == 'f8':
```

Comment:
Given that the FP8 format is gaining traction and there is a micro-scaling FP4 format, I find the naming here confusing.
Why don't we follow the naming scheme of `numpy` for types?

Reply:
Which naming scheme do you mean? NumPy does accept kind + size in bytes, unless you mean using `f` for `float32` specifically and `d` for `float64` specifically.

I think we would preferably support both, so it would be a good change to make it more flexible.

Reply:
I was thinking about the name returned by `dtype`, like `"float64"` in this example here:
```
a = np.array([1, 2], dtype=float)
a.dtype == np.dtype(np.float64)
a.dtype == np.float64
a.dtype == float
a.dtype == "float64"
a.dtype == "d"
```
But you're right, the numpy scheme for specifying types matches what we have here, so we can just keep that.

It also aligns with what is used in other places in DPCTL, so makes sense to keep it consistent.

Reply:
I do think it's sensible to support addition dtype strings ("float32", "float64") and possibly NumPy's char names ("f", "d").

It may even be sensible to expose the dpctl dtype enumerators in the dpctl namespace as enums (not dtype objects like in tensor) and add them as possible arguments, but this may be better saved for a subsequent PR. Another thought being that this could be re-used for a typed `work_group_memory` overload

Reply:
Yes, an enumeration for the possible dtypes would be useful, if we need it in multiple places, such as typed `work_group_memory` overload. But I agree, this can be done in a follow-up PR.